packages/tests/src/services/ScreenshotService.ts

```diff
@@ -0,0 +1,72 @@
+import fs = require('fs');
+import SpectreClient from 'nodeclient-spectre';
+import { ScreenshotOutput } from 'webdriver-image-comparison/build/helpers/afterScreenshot.interfaces';
+
+// TODO - .env
+const SPECTRE_API_HOST= 'http://spectre.linkerd:4140';
+const PROJECT = 'Big Design';
+const SUITE = 'Components';
+
+export default class ScreenshotService {
+  runId: number
+  browser: WebdriverIO.BrowserObject
+  spectreClient: SpectreClient
+
+  // initialize
+  constructor() {
+    this.spectreClient = new SpectreClient(SPECTRE_API_HOST)
+  }
+
+  compareElement = async (selector:string, name:string, runId:number, browser: WebdriverIO.BrowserObject): Promise<any> => {
+    this.browser = browser
+    const results = this.takeImage(selector, name)
+
+    const imageName = `${results.path}/${results.fileName}`
+    const screenshot = this.readImage(imageName)
+
+    return await this.postImage(screenshot, name, runId)
+  }
+
+  // TODO - is runId a string or a number?
+  private instantiateRun = async (): Promise<number> => {
+      const response = await this.spectreClient.createTestrun(PROJECT, SUITE)
+
+      return response.id
+  }
+
+  private takeImage = (selector:string, name:string): ScreenshotOutput => {
+    this.browser.$(selector).waitForDisplayed();
+
+    // TODO - Make sure TS definitions are added to browser
```

Comment:
Curious, what's the issue here? we should be aiming to avoid the use of ts-ignore

Reply:
browser doesn't have a ts definition for saveElement.  It'll need some investment to make it happen.  I figured we can clean that up later if this is the route taken.

Reply:
I see. Then may I suggest `return (this.browser as any).saveElement($(selector), name, { /* some options*/ })`

This basically casts `this.browser`, just for this line, to be the type `any`, which is the most generic type in TS. I think then you should be able to remove the ts-ignore. It's still not ideal but a bit better than disabling TS altogether IMO

Another suggestion I would have is, there is also a way to "augment" the type definition and merge in whatever else you need. Might be worthwhile if you'll be using `saveElement` a lot

Reply:
There is a way to add commands to browser.  I just didn't want to dig into it for a proof of concept.  I'll make the change over to `this.browser as any` and make sure the TODO is still in place

Reply:
✅